File: Dsa-day157.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int n_str = strs.size();
        int cnt = 0;
        
        int str_length = strs[0].size();
        vector<bool> sorted(n_str - 1, false);
        int j = 0;
        while(j < str_length) {
            vector<bool> current_sorted(n_str - 1, false);
            bool to_delete = false;
            for (int i = 1; i < n_str; i++) {
                if (strs[i - 1][j] > strs[i][j]) {
                    if (sorted[i] == false) {
                        to_delete = true;
                        cnt++;
                        break;
                    }
                } else if (strs[i - 1][j] < strs[i][j]) {
                    current_sorted[i] = true;
                }
            }
            if (to_delete == false) {
                bool all_sorted = true;
                for (int i = 1; i < n_str; i++) {
                    if (sorted[i] == false) {
                        sorted[i] = current_sorted[i];
                        all_sorted = false;
                    }
                }
                if (all_sorted) {
                    return cnt;
                }
            }
            j++;
        }

        return cnt;

    }
};

File: Dsa-day157.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        n, m = len(strs), len(strs[0])
        delete_count = 0
        is_sorted = [False] * (n - 1)  

        for col in range(m):
            remove_column = False
            for row in range(1, n):
                if not is_sorted[row - 1] and strs[row][col] < strs[row - 1][col]:
                    remove_column = True
                    break
            
            if remove_column:
                delete_count += 1
            else:
                for row in range(1, n):
                    if strs[row][col] > strs[row - 1][col]:
                        is_sorted[row - 1] = True

        return delete_count